### PR TITLE
Optional saving of local geometry

### LINKED ISSUE #139: Make the writing of the geometry optional.
Currently writes the global and local geometry.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
closes #139 
The geometry constructor was saving the local domain for each pe, useful for debugging but cumbersome the rest of the time. It is now added as an option, the default does not save the local geometry.

### CODE REVIEW

File: src/Geometry/soca_geom_mod.F90 [hunk 1]
```diff
@@ -68,7 +70,21 @@ subroutine geom_init(self, c_conf)
...
+
     call soca_geom_init(self%G, self%GV, self%ice_column, c_conf)
+
+    ! Set output option for local geometry
+    self%save_local_domain = .false.
+    if (config_element_exists(c_conf,"save_local_domain")) then
+       isave = config_get_int(c_conf,"read_from_file")
+       if (isave.eq.1 ) then
+          self%save_local_domain = .true.
+       else
```
Comment (aerorahul) [inline src/Geometry/soca_geom_mod.F90]: Default is set to `.false.`  An else block is not needed.

File: src/Geometry/soca_geom_mod.F90 [hunk 2]
```diff
@@ -294,22 +310,24 @@ subroutine geom_infotofile(self)
...
+    if (self%save_local_domain) then
+       ! Save local compute grid
+       pe = f_comm%rank()
+
+       write (strpe,fmt) pe
+       geom_output_pe='geom_output_'//trim(strpe)//'.nc'
+
+       call geom_get_domain_indices(self, "compute", is, ie, js, je)
+       ns = (ie - is + 1) * (je - js + 1 )
+       varname='shoremask'
```
Comment (aerorahul) [inline src/Geometry/soca_geom_mod.F90]: Can we not just remove varname? The write2pe has self%varname in it anyways.

e.g.
`call write2pe(reshape(self%shoremask,(/ns/)),'shoremask',geom_output_pe,.false.)`

File: src/Geometry/soca_geom_mod.F90 [hunk 3]
```diff
@@ -272,8 +288,8 @@ end subroutine geom_validindex
   ! ------------------------------------------------------------------------------
   !> Write geometry info to file
   subroutine geom_infotofile(self)
-    class(soca_geom), intent(in) :: self
-
+    class(soca_geom),  intent(in) :: self
```
Comment (aerorahul) [inline src/Geometry/soca_geom_mod.F90]: delete white space, here and L292

**Review by aerorahul — APPROVED** [review]
few comments. but looks good.

File: src/Geometry/soca_geom_mod.F90 (hunk 2 above)
Comment (guillaumevernieres) [reply src/Geometry/soca_geom_mod.F90]: We could, but we'd get a warning at compile time (string length not matching).

File: src/Geometry/soca_geom_mod.F90 (hunk 1 above)
Comment (guillaumevernieres) [reply src/Geometry/soca_geom_mod.F90]: oops ...

File: src/Geometry/soca_geom_mod.F90 (hunk 2 above)
Comment (aerorahul) [reply src/Geometry/soca_geom_mod.F90]: even if you turn L123 in soca/src/Utils/soca_utils.F90 to:
```
character(len=*), intent(in) :: varname
```

Comment (guillaumevernieres) [reply src/Geometry/soca_geom_mod.F90]: nope, that works, can't seem to see past the tip of my nose this am ...

**Review by travissluka — APPROVED** [review]

**Pull request merged**